pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto

```diff
@@ -668,6 +668,7 @@ enum ContainerState {
     CONTAINER_RUNNING = 1;
     CONTAINER_EXITED  = 2;
     CONTAINER_UNKNOWN = 3;
+    CONTAINER_OOMKILLED = 4;
```

Comment:
It is actually and I believe it's the main reason for not being included here already. However, OOM killed is a well defined state used by the kubelet itself (see kuberuntime_manager) and we can't leave this state as a string to be included in the Reason field on the container state. If that's the current idea however, let's at least make the documentation clear around this well specified reason of exited. Right now, nobody knows about this particular status from the CRI api and that's a huge lack as you're required to go to kube source code to learn about it and use it in your cri implementation.